src/vdaf/prio3.rs

```diff
@@ -862,23 +903,20 @@ where
         // Compute the joint randomness.
         let (joint_rand_seed, joint_rand_part, joint_rand) = if self.typ.joint_rand_len() > 0 {
             let mut joint_rand_part_prg = P::init(
-                msg.joint_rand_param.as_ref().unwrap().blind.as_ref(),
+                msg.joint_rand_blind.as_ref().unwrap().as_ref(),
                 &Self::custom(DST_JOINT_RAND_PART),
             );
             joint_rand_part_prg.update(&[agg_id]);
             joint_rand_part_prg.update(nonce);
-            for x in input_share {
+            for x in measurement_share {
                 joint_rand_part_prg.update(&(*x).into());
             }
-            let joint_rand_part = joint_rand_part_prg.into_seed();
-
-            let hints = &msg.joint_rand_param.as_ref().unwrap().seed_hint;
-            let joint_rand_seed = Self::derive_joint_rand_seed(
-                hints[..agg_id as usize]
-                    .iter()
-                    .chain(std::iter::once(&joint_rand_part))
-                    .chain(hints[agg_id as usize..].iter()),
-            );
+            let own_joint_rand_part = joint_rand_part_prg.into_seed();
+
+            let mut corrected_joint_rand_parts = public_share.joint_rand_parts.clone();
+            corrected_joint_rand_parts[agg_id as usize] = own_joint_rand_part.clone();
```

Comment:
We should be able to avoid these clones. Do you think doing so would be worth the effort? The downside would be less readable code I think.